tests/trezor/trezor_tests.cpp

```diff
@@ -1931,38 +2195,54 @@ bool wallet_api_tests::generate(std::vector<test_event_entry>& events)
   const auto api_net_type = m_network_type == TESTNET ? Monero::TESTNET : Monero::MAINNET;
 
   Monero::WalletManager *wmgr = Monero::WalletManagerFactory::getWalletManager();
-  std::unique_ptr<Monero::Wallet> w{wmgr->createWalletFromDevice(wallet_path, "", api_net_type, m_trezor_path, 1)};
+  std::unique_ptr<Monero::Wallet> w{wmgr->createWalletFromDevice(wallet_path, "", api_net_type, m_trezor_path, 1, "", 1, this)};
   CHECK_AND_ASSERT_THROW_MES(w->init(daemon()->rpc_addr(), 0), "Wallet init fail");
 
   auto walletImpl = dynamic_cast<Monero::WalletImpl *>(w.get());
   CHECK_AND_ASSERT_THROW_MES(walletImpl, "Dynamic wallet cast failed");
   WalletApiAccessorTest::allow_mismatched_daemon_version(walletImpl, true);
+  walletImpl->setTrustedDaemon(true);
 
   CHECK_AND_ASSERT_THROW_MES(w->refresh(), "Refresh fail");
+
+  uint64_t spent, unspent;
+  walletImpl->coldKeyImageSync(spent, unspent);
+  CHECK_AND_ASSERT_THROW_MES(walletImpl->rescanSpent(), "Rescan spent fail");
```

Comment:
This test worked previously fine by coincidence. By changing chain structure wallet API created double-spend transactions. We have to scan the wallet. 

The code now demonstrates the correct flow and tests also key image sync via API.